### PR TITLE
feat: add option for a button after checkout

### PR DESCRIPTION
**Pull request by leogermani**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Implements an optional link to click after the transaction is complete on the modal checkout.

The button supports linking to the page referrer (which falls back to the current page URL if the referrer is empty) or a custom URL.

image

### How to test the changes in this Pull Request:

1. Checkout this branch and https://github.com/Automattic/newspack-theme/pull/2180
2. Add a checkout button to a page
3. Select the option to go to the Referrer after purchase and add a button label
4. Add a link to this page in the site menu
5. Visit an article
6. Go to the page via the menu link
7. Click the button and buy the product
8. Confirm the button shows up in the thank you message and link back to the article you were seeing
9. In a new tab, copy and paste the page URL directly into the browser so you get no referrer
10. Make the purchase again and confirm the button links to the current page
11. Edit the button and choose to go to a custom URL
12. Repeat the purchase and confirm the button now takes you to that URL
13. Edit the button again and choose not to show the button
14. Purchase again and confirm there is no button in the thank you message

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/blocks/checkout-button/edit.js [hunk 1]
```diff
@@ -153,6 +153,48 @@ function ProductControl( props ) {
...
+				label={ __( 'Redirect after success', 'newspack-blocks' ) }
+				help={ __(
+					'Select whether the user should be presented with a button to navigate after a successful purchase.',
+					'newspack-blocks'
+				) }
+				value={ attributes.afterSuccessBehavior }
+				options={ [
+					{ label: __( 'Do not show a button', 'newspack-blocks' ), value: '' },
+					{ label: __( 'Go to a custom URL', 'newspack-blocks' ), value: 'custom' },
+					{ label: __( 'Go to the referrer page', 'newspack-blocks' ), value: 'referrer' },
```
Comment (adekbadek) [inline src/blocks/checkout-button/edit.js:170]: Suggestion: "previous page" might be more understandable for non-technical people.

**Review by adekbadek — CHANGES_REQUESTED** [review]
When the custom URL field is never set, the button will not be displayed. To reproduce:

1. Insert the checkout button on a page, choose "Go to referrer page" option
2. Observe that the button is not rendered after checkout
3. Back in the editor, choose "Go to a custom URL" option and fill in a URL. Save and edit again, choosing "Go to referrer page". Save again.
4. Observe the button is rendered after checkout

**Comment by miguelpeixe** [comment]
Do you think we should render the button linked to the current page when the URL is not filled?

**Comment by adekbadek** [comment]
I think that without the URL set, the button should just close the modal. But the issue I'm describing is about something else: according to the testing instructions, if the URL is not filled and "Go to referrer page" is set, the button should be rendered and link to the referrer. And it's not, under conditions described in my review.

File: src/blocks/checkout-button/edit.js (hunk 1 above)
Comment (miguelpeixe) [reply src/blocks/checkout-button/edit.js]: Updated in 4325f94b4f83c785cc64bd00c164b8338eae766f

**Comment by miguelpeixe** [comment]
Ah, sorry for the confusion. I was able to reproduce the issue.

4c514ae272b2f8fe437d9d0b591bec8d9836081d changes the block save to render the URL input if the `afterSuccessBehavior` is set, fixing the issue. I'm not rendering the inputs at all times because it causes a markup mismatch for existing blocks, requiring "attempt block recovery".

File: src/blocks/checkout-button/edit.js [hunk 2]
```diff
@@ -153,6 +153,48 @@ function ProductControl( props ) {
 	);
 }
 
+function RedirectAfterSuccess( props ) {
+	const { attributes, setAttributes } = props;
+	return (
+		<>
+			<SelectControl
+				label={ __( 'Redirect after success', 'newspack-blocks' ) }
```
Comment (adekbadek) [inline src/blocks/checkout-button/edit.js:161]: Suggestion: this wording makes me think that there will be an automatic page redirection after the purchase is completed. Shouldn't this label inform the user about the post-checkout _button_ instead?

**Review by adekbadek — APPROVED** [review]
Left a wording suggestion.

File: src/blocks/checkout-button/edit.js (hunk 2 above)
Comment (miguelpeixe) [reply src/blocks/checkout-button/edit.js]: Updated in 518e0c32d3b971b4ad144560a2f457a71423f4d6

**Pull request merged**